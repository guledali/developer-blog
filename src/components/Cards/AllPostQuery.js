import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import Cards from './Cards'

const AllPost = ({ children }) => (
  <StaticQuery
    query={graphql`
      {
        dato {
          allArticles {
            excerpt
            image {
              url
            }
            id
            title
            _publishedAt
          }
        }
      }
    `}
    render={data =>
      data.dato.allArticles.map((post, i) => <Cards cards={post} key={i} />)
    }
  />
)

export default AllPost
