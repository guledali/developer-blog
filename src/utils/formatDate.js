export default function formatDate(args) {
  return args.slice(0, 10)
}
