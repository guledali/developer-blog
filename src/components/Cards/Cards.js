import React, { Component } from 'react'
import {
  Card,
  Button,
  CardImg,
  CardTitle,
  CardText,
  CardBody,
  Col,
} from 'reactstrap'
import formatDate from '../../utils/formatDate'

class Cards extends Component {
  render() {
    return (
      <Col lg={4}>
        <Card className="mt-2">
          <CardImg
            style={{ background: '#000' }}
            top
            width="100%"
            src={this.props.image}
            alt="Card image cap"
          />
          <CardBody>
            <CardTitle className="font-weight-bold">
              {this.props.title}
            </CardTitle>
            <CardText>{this.props.short}</CardText>
            <CardText>
              <small className="text-muted">
                Published at {formatDate(this.props.date)}
              </small>
            </CardText>
            <>
              <style type="text/css">
                {`
                .btn-warning {
                  background-color: #f5da55;
                  border-color: #f5da55;
                  color: #323330
                }
              `}
              </style>
              <Button color="warning" className="w-100">
                Read more
              </Button>
            </>
          </CardBody>
        </Card>
      </Col>
    )
  }
}

export default Cards
