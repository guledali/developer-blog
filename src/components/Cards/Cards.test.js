import React from 'react'
import renderer from 'react-test-renderer'
import Cards from './Cards'
import formatDate from '../../utils/formatDate'

//title={post.title}
//image={post.image.url}
//date={post._createdAt}
//short={post.excerpt}
//id={post.id}
//key={i}
describe('<Cards />', () => {
  it('Display component with data', () => {
    const data = {
      data: {
        dato: {
          allArticles: {
            excerpt:
              'When PHP Framework Sucks Series: How not to shape your app in the shape of the framework',
            image: {
              url:
                'https://www.datocms-assets.com/8778/1546791237-php-example.png',
            },
            id: '592737',
            title: 'Part of "When PHP Framework Sucks" series',
            _createdAt: '2018-12-21T14:07:09.256Z',
          },
        },
      },
    }
    const tree = renderer
      .create(
        <Cards
          title={data.data.dato.allArticles.title}
          image={data.data.dato.allArticles.image.url}
          date={formatDate(data.data.dato.allArticles._createdAt)}
          short={data.data.dato.allArticles.excerpt}
          id={data.data.dato.allArticles.id}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
