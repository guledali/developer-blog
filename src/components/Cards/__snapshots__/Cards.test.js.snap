// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards /> Display component with data 1`] = `
<div
  className="col-lg-4"
>
  <div
    className="mt-2 card"
  >
    <img
      alt="Card image cap"
      className="card-img-top"
      src="https://www.datocms-assets.com/8778/1546791237-php-example.png"
      style={
        Object {
          "background": "#000",
        }
      }
      width="100%"
    />
    <div
      className="card-body"
    >
      <div
        className="font-weight-bold card-title"
      >
        Part of "When PHP Framework Sucks" series
      </div>
      <p
        className="card-text"
      >
        When PHP Framework Sucks Series: How not to shape your app in the shape of the framework
      </p>
      <p
        className="card-text"
      >
        <small
          className="text-muted"
        >
          Published at 
          2018-12-21
        </small>
      </p>
      <style
        type="text/css"
      >
        
                .btn-warning {
                  background-color: #f5da55;
                  border-color: #f5da55;
                  color: #323330
                }
              
      </style>
      <button
        aria-label={null}
        className="w-100 btn btn-warning"
        onClick={[Function]}
      >
        <a
          className="text-dark text-decoration-none"
          href="/SinglePost/592737"
        >
          Read more
        </a>
      </button>
    </div>
  </div>
</div>
`;
